reana/reana_benchmark/analyze.py

```diff
@@ -210,7 +242,17 @@ def _build_execution_progress_plot(
     # force integers on y axis
     ax.yaxis.set_major_locator(MaxNLocator(integer=True))
 
-    ax.xaxis.set_major_locator(mdates.MinuteLocator(interval=interval))
+    # set x axis interval and ticks
+    interval = plot_parameters.get("time_interval")
+    if not interval:
+        interval = _approximate_locator_interval_seconds(df)
+    ax.xaxis.set_major_locator(mdates.SecondLocator(interval=interval))
+    if not plot_parameters.get("time_interval"):
+        locs, _ = plt.xticks()
+        additional_tick = (locs[1] - locs[0]) + locs[-1]
+        locs = np.append(locs, additional_tick)
+        ax.set_xticks(locs)
```

Comment:
*question:* What do you think about adding an additional tick? Sometimes even if the interval is approximated, there won't be enough ticks to cover all points/lines.

Reply:
Currently a tick to mark the beginning is missing. Before we could see when workflows have started
Before
![hw_execution_progress_1_10 copy 2](https://user-images.githubusercontent.com/4990025/144247687-1c313ae1-c35e-4b06-9a9d-b8f912a0776d.png)
Now:
![hw_execution_progress_1_10](https://user-images.githubusercontent.com/4990025/144247715-6daeb418-e1b6-4718-bb04-933708b1eacc.png)

Reply:
Actually, do we even need to approximate the interval by ourselves at all? If this peace of code is commented out (lines 245-257), `matplotlib` will do it for us:
![hw_execution_progress_1_10](https://user-images.githubusercontent.com/4990025/144249251-48b986b0-0f91-4105-b96c-d60f4270127e.png)

Reply:
Check the discussions in the issue. I would prefer to keep it manual and specify `-i` but this is what was decided. 

*suggestion:*: following on your observation, we can just get rid of most of the PR code and make `-i` just optional instead of the default value. Then matplotlib will do all stuff for us. Sounds nice. Let me test it.

Reply:
yeah, that's what I meant, perhaps it would be enough to do:
```python
interval = plot_parameters.get("time_interval")
if interval:
    ax.xaxis.set_major_locator(mdates.MinuteLocator(interval=interval))
  ```

Reply:
I tested the above simpler version and it still has some flaws (some ticks are missing at the beginning or end) but not much difference compared to a big chunk of code I wrote. I think it should solve the main issue. Thank you for the suggestion. I got too focused on one solution and didn't realize to check the simpler one. Good to have a second pair of eyes.

Example of "problematic" one with the above simple solution:

![histogram_execution_progress_1_5](https://user-images.githubusercontent.com/15694232/144320057-7cf65d93-a078-44b3-8c37-9fae2c1fa75d.png)

Reply:
I put a second commit with a simpler version to keep the older variant while PR is in review. If we will go with a simpler one I will squash commits at the end.